src/common/transformations/src/transformations/op_conversions/convert_ti_to_sequences.cpp

```diff
@@ -830,6 +831,291 @@ ov::pass::ConvertLoopToLSTMSequence::ConvertLoopToLSTMSequence() {
+        int64_t counter_step = -1;
+        if (!get_scalar_constant_value(condition_map.at(counter_step_label), counter_step) || counter_step != 1) {
+            return false;
+        }
+        // get initial value of counter
+        int64_t initial_counter = -1;
+        auto counter_param =
+            ov::as_type_ptr<ov::op::v0::Parameter>(condition_map.at(counter_label).get_node_shared_ptr());
+        if (!counter_param) {
+            return false;
+        }
+        for (const auto& input_desc : loop->get_input_descriptions()) {
+            auto body_param_idx = input_desc->m_body_parameter_index;
+            auto input_idx = input_desc->m_input_index;
+            if (body_params[body_param_idx] != counter_param) {
+                continue;
+            }
+            // it must be merged input and incremented each iteration
+            auto merged_desc = ov::as_type_ptr<ov::op::util::MultiSubGraphOp::MergedInputDescription>(input_desc);
+            if (!merged_desc) {
+                return false;
+            }
+            auto result_idx = merged_desc->m_body_value_index;
+            auto update_counter = body_results[result_idx]->input_value(0).get_node_shared_ptr();
+            if (update_counter != condition_map.at(updated_counter_label).get_node_shared_ptr()) {
+                return false;
+            }
+            // get initial value of counter
+            if (!get_scalar_constant_value(loop->input_value(input_idx), initial_counter)) {
+                return false;
+            }
+            // suitable counter-parameter is found and checked
+            break;
+        }
+        if (initial_counter != 0) {
+            return false;
+        }
+        // retrieve a number of iterations
+        int64_t num_iters = -1;
+        ov::Output<ov::Node> num_iters_output;
+        if (condition_map.count(num_iter_param_label)) {
+            auto num_iter_param = condition_map.at(num_iter_param_label).get_node_shared_ptr();
+            for (const auto& input_desc : loop->get_input_descriptions()) {
+                auto body_param_idx = input_desc->m_body_parameter_index;
+                auto input_idx = input_desc->m_input_index;
+                if (body_params[body_param_idx] != num_iter_param) {
+                    continue;
+                }
+                num_iters_output = loop->input_value(input_idx);
+                break;
+            }
+        } else if (condition_map.count(num_iter_const_label)) {
+            if (!get_scalar_constant_value(condition_map.at(num_iter_const_label), num_iters) || num_iters < 1) {
+                return false;
+            }
+            num_iters_output = std::make_shared<ov::op::v0::Constant>(element::i64, Shape{}, num_iters);
+        } else {
+            return false;
+        }
+
+        // check that body-graph contains a pattern corresponding LSTMCell
+        ov::pass::pattern::Matcher lstm_cell_matcher(result_hidden_state_label);
```

Comment:
is it possible to move "LSTMCell pattern " and related code to a separate function?

Reply:
Done